src/plonk/eval.rs

```diff
@@ -178,14 +189,21 @@ impl<F: PrimeField, E: GetDataForEval<F>> Eval<F, MultiPolynomial<F>> for E {
 }
 
 impl<F: PrimeField, E: GetDataForEval<F>> Eval<F, GroupedPoly<F>> for E {
-    fn eval(&self, _poly: &GroupedPoly<F>, _row_index: usize) -> Result<F, Error> {
-        todo!("#159")
+    type Output = Box<[F]>;
```

Comment:
Do we really need to make the trait more complex in order to impl Eval trait to GroupedPoly?

Reply:
Read the description, I'm thinking of redoing this part later in full, rethinking the eval, since this trait is not even used in nifs, as it is.

However, in order not to solve this problem right now I suggest to postpone it and focus on MPV milestone

Reply:
It's not that it's an extensive task, I just want to understand how this part will look like with protogalaxy, so that I don't have to redo it twice

Reply:
GroupedPoly is **just** a vector of Expression and nothing more. We don't need define a new method and modify the trait just for evaluate a vector of Expression given that we already have eval method for Expression. It's not worth it.

Reply:
There is no point to modify the Eval trait with your current modification

Reply:
If you insist, I will redo this trait right away. In fact it's not really needed in this form, as the final use case in nifs is not covered by this API.

In other words, isolation of the GraphEvaluator interface is not achieved. And that's the purpose of this trait

Reply:
Yeah, please change the trait back (with only single generic F). and just treat GroupedPoly as vector of Expression.

Reply:
You misunderstand me, I'm not going to change it back, as its main task of “isolating the evaluation interface” it no longer performs, as the `GraphEvaluator` is separately created for expr (and this operation costs resources) and then executes for each line separately. The old design would just add a big overhead (because of that, [here ](https://github.com/snarkify/sirius/blob/29d3df288a9a7bcb762b3927a35b6487e56638a4/src/nifs/vanilla.rs#L106) this trait not used at all)

I will submit a PR to this branch with a different API design

Reply:
Could you prepare the design doc and let's review it first.

Reply:
https://github.com/snarkify/sirius/issues/244